azurerm/internal/services/netapp/resource_arm_netapp_volume.go

```diff
@@ -122,7 +134,7 @@ func resourceArmNetAppVolume() *schema.Resource {
 							Type:     schema.TypeBool,
 							Required: true,
 						},
-						"nfsv4_enabled": {
+						"nfsv41_enabled": {
```

Comment:
What is the point of these enabled functions if we have the protocol property ? if that property is only cifs, can nfsv3 still be enabled?

For that matter can you enabled cifs and nfs3 for the same export rule?

Reply:
Also by the nature of changing the name of this property the hashcode will change and a state migration will be required

Reply:
For the first question, these properties are specific to the NFS export policy and needs to be separately defined as specified by the APIs. 

For the second question, as I commented above, we will have this support as soon as ANF gets support for multiprotocol, and yes, you can define as of now cifs + nfsv3, cifs + nfsv4 but not nfsv3 + nfsv4 on the export policy, the latter will cause a 500 response. But even if the multiprotocol combinations as of today are not returning an error, it doesn't mean it will work as expected since we didn't add support for it yet.

Regarding the last comment of this conversation, not sure if I follow what this comment mean, I very new to Terraform in general and this is my first contribution :-). Can you provide more details on what does that mean?

Reply:
So because there is a hashcode created for each element of this list, adding a new property will change the computed hashcode. So going from version 1.41-1.42 with the same config will produce a diff.

We could also simplify this to a `protocols_enabled` property instead so in to future we don't have to worry about adding a new property and doing another state migration?